test/es-module/test-esm-detect-ambiguous.mjs

```diff
@@ -258,9 +261,12 @@ describe('Module syntax detection', { concurrency: !process.env.TEST_PARALLEL },
         fixtures.path('es-modules/tla/unresolved.js'),
       ]);
 
-      strictEqual(stderr, '');
+      match(
+        stderr,
+        /SyntaxError: Top-level await is not supported in CommonJS modules\. To use top-level await, add "type": "module" to your package\.json or rename the file to use the \.mjs extension/
```

Comment:
This test is titled “reports unfinished top-level `await`”. The code here _should_ be parsed and run as ESM, and it should error with exit code 13 like the original test. This isn’t a test that should need updating as part of the goals of this PR.